slingr-stack/microsoft-sharepoint-package
# Feature/#3 refactor sharepoint package

Pull-request for issue https://github.com/slingr-stack/microsoft-sharepoint-package/issues/3 created by @agreggio-slingr

## What it does?

Implement the OAuth package.

## How to test it?

For testing purposes, you need to have Azure credentials configured. Create a test application to perform these tests.
Once the previous step is completed, open builder and runtime test1, simply execute the following commands from the console:

pkg.sharepoint.functions.getAccessToken();

This will allow you to grant permissions to the Slingr application to operate on behalf of your user (read-only).

log(JSON.stringify(pkg.sharepoint.functions.sites.get()));

This will make a query to the SharePoint API, and you should expect to receive the following response:

{"@odata.context":"https://graph.microsoft.com/v1.0/$metadata#sites","value":[]}

## Implementation notes

NA

## Deployment notes

There are no deployment notes
